GameWorld/HexGrid/DrawableHexSphere.cs

```diff
@@ -114,6 +122,46 @@ public void SetGroundHeight(int groundHeight)
             InitializeVertices();
         }
 
+        public double GetHeightFromCoveragePercent(int coveragePercent)
+        {
+            double resultHeight = 0.5f;
+            List<double> tiles_heights = new List<double>();
+
+            foreach (var t in _sphereGrid.Tiles)
+            {
+                tiles_heights.Add(t.Height);
+            }
+
+            tiles_heights.Sort();
+
+            if (0 > coveragePercent || 100 < coveragePercent)
+            {
+                resultHeight = 0.5f;
+            }
+            else if (0 == coveragePercent)
+            {
+                resultHeight = 0.0f;
+            }
+            else if (100 == coveragePercent)
+            {
+                resultHeight = 1.0f;
+            }
+            else
+            {
+                int index = (tiles_heights.Count * coveragePercent / 100) - 1;
+                if (0 < index)
```

Comment:
A case is possible where Count of tiles would be lower than 100 in this case 1 percent coverage would result in index of -1 which would case fatal error